### PR TITLE
Track InheritedElement dependencies in diagnostic properties

### PR DESCRIPTION
**Pull request by jonahwilliams**
Information can be surfaced from command line or widget inspector. Allows users to see which widgets are currently being depended on via a `.of(context)` or similar InheritedWidget pattern.

Some work towards https://github.com/flutter/flutter/issues/27100

### CODE REVIEW

**Comment by goderbauer** [comment]
Can you add a dedicated test for this? If we ever decided to delete the Table widget, this would be untested :)

**Comment by jonahwilliams** [comment]
A fair point :)

File: packages/flutter/test/widgets/inherited_dependencies.dart [hunk 1]
```diff
@@ -0,0 +1,45 @@
+import 'package:flutter/src/widgets/basic.dart';
```
Comment (goderbauer) [inline packages/flutter/test/widgets/inherited_dependencies.dart]: Missing license header.

**Review by goderbauer — APPROVED** [review]
LGTM

File: packages/flutter/test/widgets/inherited_dependencies.dart (hunk 1 above)
Comment (jonahwilliams) [reply packages/flutter/test/widgets/inherited_dependencies.dart]: Oops fixed

File: packages/flutter/test/widgets/inherited_dependencies.dart [hunk 2]
```diff
@@ -0,0 +1,49 @@
+// Copyright 2019 The Chromium Authors. All rights reserved.
```
Comment (goderbauer) [inline packages/flutter/test/widgets/inherited_dependencies.dart]: One more: shouldn't the file name end in `_test.dart`?

Comment (jonahwilliams) [reply packages/flutter/test/widgets/inherited_dependencies.dart]: Uh D'Oh

**Pull request merged**